app/src/androidTest/java/com/monkeyteam/chimpagne/FindEventScreenTest.kt

```diff
@@ -91,7 +107,50 @@ class FindEventScreenTest {
 
     composeTestRule.onNodeWithTag("join_button").performClick()
 
+    // Should be false because the user is not logged in (guest) so it will not trigger the
+    // joinEvent
+    // function hence no uiState will load anything
+    assertFalse(findViewModel.uiState.value.loading)
+  }
+
+  @OptIn(ExperimentalMaterial3Api::class)
+  @Test
+  fun testJoinEventFunctionalityUserNotInEvent() {
+    val database = Database()
+    val findViewModel = FindEventsViewModel(database)
+    val accountViewModel = AccountViewModel(database)
+    val sampleEvent =
+        ChimpagneEvent(id = "sample123", title = "Sample Event", description = "Sample Description")
+    findViewModel.setResultEvents(mapOf(sampleEvent.id to sampleEvent))
+
+    composeTestRule.setContent {
+      val navController = rememberNavController()
+      val navActions = NavigationActions(navController)
+
+      FindEventMapScreen({}, findViewModel, accountViewModel, navActions)
+    }
+
+    accountManager.signInTo(anAccount)
+    val n = sampleEvent.getRole(anAccount.firebaseAuthUID)
+    if (n == ChimpagneRole.GUEST) {
```

Comment:
A when should be used here:
https://kotlinlang.org/docs/control-flow.html#when-expression

Reply:
Deleted the logs, they were not necessary, I had forgotten to remove them :)